### PR TITLE
REV-38-link user roles with backend

### PR DESCRIPTION
**Pull request by IbrahimAldereni-socuimtech**
### Description
- link user roles with backend
- backend PR: 
https://github.com/abukhadraop-or/reverifi-api/pull/14

### CODE REVIEW

File: src/services/user.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { apiUrl } from 'config/config';
+import http from 'utils/http';
+
+/**
+ * Service that update user roles
+ *
+ * @param {Array(String)} rolesIds roles ids array
```
Comment (Mahmoudzaid2994) [inline src/services/user.js:7]: add new line please

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,3 +1,8 @@
 {
-    "prettier.trailingComma": "all"
+  "prettier.trailingComma": "all",
```
Comment (AhmadAlShwaiki) [inline .vscode/settings.json:2]: Please revert the above and rebase once this [PR](https://github.com/abukhadraop-or/reverifi/pull/54) is merged

File: src/components/my-roles/RoleCard/index.jsx [hunk 3]
```diff
@@ -12,23 +12,33 @@ import { Card, CheckIcon, Overview, Rectangle } from './role-card.styles';
  * @returns {JSX.Element} Role Card holding role details.
  */
 function RoleCard({ data, setSelectedRole }) {
-  const { Icon, overview, role } = data;
+  const { Icon, overview, role, id, dbRoles } = data;
```
Comment (AhmadAlShwaiki) [inline src/components/my-roles/RoleCard/index.jsx:15]:
```suggestion
  const { Icon, overview, role, id, dbRoles:selectedRoles } = data;
```
```suggestion
  const { Icon, overview, role, id, dbRoles } = data;
```

File: src/components/my-roles/RoleCardsContainer/data.js [hunk 4]
```diff
@@ -13,55 +13,39 @@ const rolesData = [
   {
     Icon: Icon(seller),
     overview: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
-    role: 'Seller',
   },
```
Comment (AhmadAlShwaiki) [inline src/components/my-roles/RoleCardsContainer/data.js:16]:
```suggestion
    selected: true
  },
```

File: src/components/my-roles/Roles/index.jsx [hunk 5]
```diff
@@ -1,28 +1,35 @@
...
  * @returns Container aligning myRoles page components.
  */
 function Roles() {
-  // eslint-disable-next-line no-unused-vars
   const [selectedRoles, setSelectedRoles] = useState(() => new Set());
 
+  /**
+   * handle start process button click
+   */
+  const startProcessHandler = async () => {
```
Comment (AhmadAlShwaiki) [inline src/components/my-roles/Roles/index.jsx:18]:
```suggestion
  const handleOnClick = async () => {
```

**Review by AhmadAlShwaiki — CHANGES_REQUESTED** [review]
Looks really good, just few tweaks

File: src/components/my-roles/RoleCardsContainer/index.jsx [hunk 6]
```diff
@@ -14,24 +19,33 @@ import { CardsContainer } from './role-cards-container.styles';
  */
 function RoleCardsContainer({ setSelectedRoles }) {
   const [roles, setRoles] = useState([]);
+  const [dbRoles, setDbRoles] = useState([]);
```
Comment (AhmadAlShwaiki) [inline src/components/my-roles/RoleCardsContainer/index.jsx:22]: Please adjust here as well, the name isn't appropriate.  

https://github.com/abukhadraop-or/reverifi/pull/52/files#diff-d141e3e38da901f992e4c9af91db349c4137ec5f2571227854f6932f4d8a757fR15

**Review by AhmadAlShwaiki — APPROVED** [review]
Looks good!

**Pull request merged**